module/documents/active-effect.mjs

```diff
@@ -354,7 +354,83 @@ export default class ActiveEffect5e extends ActiveEffect {
   }
 
   /* -------------------------------------------- */
-  /*  Exhaustion Handling                         */
+  /*  Concentration Handling                      */
+  /* -------------------------------------------- */
+
+  /**
+   * Create an effect for concentration on an actor, optionally replacing an existing effect.
+   * @param {Item5e} item                         The item on which to begin concentrating.
+   * @param {ActiveEffect5e} [existing]           An existing effect to replace.
+   * @param {object} [options]                    Additional data provided for the effect instance.
+   * @returns {Promise<ActiveEffect5e|null>}      A promise that resolves to the created effect.
+   */
+  static async createConcentrationEffect(item, existing=null, options={}) {
+    if ( !item.isEmbedded || !item.requiresConcentration ) {
+      throw new Error("You may not begin concentration on this item!");
+    }
+
+    const actor = item.actor;
+
+    const baseData = foundry.utils.mergeObject({
+      name: `${game.i18n.localize("DND5E.Concentration")}: ${item.name}`,
+      description: game.i18n.format("DND5E.ConcentratingOn", {
+        name: item.name,
+        type: game.i18n.localize(`TYPES.Item.${item.type}`)
+      }),
+      duration: ActiveEffect5e._getEffectDuration(item),
+      "flags.dnd5e.itemData": actor.items.has(item.id) ? item.id : item.toObject(),
+      origin: item.uuid
+    }, options);
+
+    const effects = actor.concentration?.effects ?? new Set();
+    existing ??= effects.find(e => {
+      const d = e.flags.dnd5e?.itemData ?? {};
+      return (d === item.id) || (d._id === item.id);
+    });
+
+    let effectData;
+    // Replace an existing effect if already concentrating on this item or if another effect has been chosen.
+    if ( existing ) {
+      const data = existing.flags.dnd5e?.itemData ?? {};
+      const sameItem = (data === item.id) || (data._id === item.id);
+      if ( sameItem ) {
+        effectData = foundry.utils.mergeObject({...existing.toObject(), duration: {}}, baseData);
```

Comment:
I don't think I quite understand this logic. What do we gain from inheriting the existing effect's data vs. just creating a new effect that would be (seemingly) the same as the existing one if the origin Items are the same too, as we're asserting that they are here?